core/runtime/binaryen/runtime_environment.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_RUNTIME_BINARYEN_RUNTIME_ENVIRONMENT
+#define KAGOME_CORE_RUNTIME_BINARYEN_RUNTIME_ENVIRONMENT
+
+#include <memory>
+
+#include <boost/optional.hpp>
+
+#include "outcome/outcome.hpp"
+
+namespace kagome::common {
+  class Buffer;
+}
+
+namespace kagome::storage::trie {
+  class TopperTrieBatch;
+  class TrieBatch;
+}  // namespace kagome::storage::trie
+
+namespace kagome::runtime {
+  class WasmMemory;
+}
+
+namespace kagome::runtime::binaryen {
+  class RuntimeExternalInterface;
+  class WasmModuleInstance;
+  class WasmModule;
+
+  /**
+   * Runtime environment is a structure that contains data necessary to operate
+   * with Runtime: memory, WASM module, storage batch
+   */
+  class RuntimeEnvironment {
+   public:
+    static outcome::result<RuntimeEnvironment> create(
+        const std::shared_ptr<RuntimeExternalInterface> &rei,
+        const std::shared_ptr<WasmModule> &module,
+        const common::Buffer &state_code);
+
+    RuntimeEnvironment(RuntimeEnvironment &&) = default;
+    RuntimeEnvironment &operator=(RuntimeEnvironment &&) = default;
+
+    RuntimeEnvironment(const RuntimeEnvironment &) = delete;
+    RuntimeEnvironment &operator=(const RuntimeEnvironment &) = delete;
+
+    ~RuntimeEnvironment() = default;
```

Comment:
```suggestion
```
default destructor is not required to define explicitly as soon as it is not virtual

Reply:
The same here, in addition I initially forgot to do that and clang-tidy complained.